docs/TypeScript.md

```diff
@@ -128,250 +128,98 @@ The type system heavily relies on generic properties to provide the most accurat
-With typebox you define your schema within your code and use them directly as types or schemas as you need them.
+- `typebox` - [github](https://github.com/sinclairzx81/typebox)
+- `json-schema-to-ts` - [github](https://github.com/ThomasAribart/json-schema-to-ts)
 
-When you want to use it for validation of some payload in a fastify route you can do it as follows:
+#### TypeBox
 
-1. Install `typebox` in your project.
+The following sets up a TypeBox Type Provider
 
-    ```bash
-    npm i @sinclair/typebox
-    ```
-
-2. Define the schema you need with `Type` and create the respective type  with `Static`.
-  
-    ```typescript
-    import { Static, Type } from '@sinclair/typebox'
-
-    const User = Type.Object({
-      name: Type.String(),
-      mail: Type.Optional(Type.String({ format: "email" })),
-    });
-    type UserType = Static<typeof User>;
-    ```
-
-3. Use the defined type and schema during the definition of your route
-
-    ```typescript
-    const app = fastify();
-
-    app.post<{ Body: UserType; Reply: UserType }>(
-      "/",
-      {
-        schema: {
-          body: User,
-          response: {
-            200: User,
-          },
-        },
-      },
-      (req, rep) => {
-        const { body: user } = req;
-        /* user has type
-        * const user: StaticProperties<{
-        *  name: TString;
-        *  mail: TOptional<TString>;
-        * }>
-        */
-        //...
-        rep.status(200).send(user);
-      }
-    );
-    ```
-
-#### Schemas in JSON Files
-
-In the last example we used interfaces to define the types for the request querystring and headers. Many users will already be using JSON Schemas to define these properties, and luckily there is a way to transform existing JSON Schemas into TypeScript interfaces!
-
-1. If you did not complete the 'Getting Started' example, go back and follow steps 1-4 first.
-2. Install the `json-schema-to-typescript` module:
-
-   ```bash
-   npm i -D json-schema-to-typescript
-   ```
-
-3. Create a new folder called `schemas` and add two files `headers.json` and `querystring.json`. Copy and paste the following schema definitions into the respective files:
-
-   ```json
-   {
-     "title": "Headers Schema",
-     "type": "object",
-     "properties": {
-       "h-Custom": { "type": "string" }
-     },
-     "additionalProperties": false,
-     "required": ["h-Custom"]
-   }
-   ```
-
-   ```json
-   {
-     "title": "Querystring Schema",
-     "type": "object",
-     "properties": {
-       "username": { "type": "string" },
-       "password": { "type": "string" }
-     },
-     "additionalProperties": false,
-     "required": ["username", "password"]
-   }
-   ```
-
-4. Add a `compile-schemas` script to the package.json:
-
-```json
-   {
-     "scripts": {
-       "compile-schemas": "json2ts -i schemas -o types"
-     }
-   }
+```bash
+$ npm install fastify-type-provider-typebox --save
```

Comment:
```suggestion
$ npm install @fastify/typebox-type-provider --save
```

The repositories weren't created yet, so it might change.

Reply:
I think it was based on this comment: https://github.com/fastify/fastify/pull/3398#discussion_r738561143 but as you said, the packages are not available yet – so I don't know if it makes sense to keep this or update it to use `@fastify` namespace.